### PR TITLE
Luzon new processes

### PR DESCRIPTION
**Pull request by glorialuzon**
glorialuzon Ok: 5 [](https://gitlab.cern.ch/rest-for-physics/geant4lib/-/commits/luzon-new-processes) [](https://gitlab.cern.ch/rest-for-physics/restG4/-/commits/luzon-new-processes) [](https://gitlab.cern.ch/rest-for-physics/framework/-/commits/luzon-new-processes)

Upgrading the RadioactiveDecay Processes.
Two files TRestGeant4Track.cxx and TRestGeant4Track.h have been modified to add the RadioactiveDecayBase and the Radioactivation processes.  The old RadioactiveDecay Process has been kept.

### CODE REVIEW

**Comment by jgalan** [comment]
Not important right now, but the description at the PR should be updated.

@juanangp @lobis clang-formatting is only active for framework repository I guess.

**Review by jgalan — APPROVED** [review]

**Comment by lobis** [comment]
> Not important right now, but the description at the PR should be updated.
> 
> @juanangp @lobis clang-formatting is only active for framework repository I guess.

Yes, I was just wondering this, I think @juanangp set it up to work with submodules too, maybe we should take a look.

**Comment by juanangp** [comment]
> > Not important right now, but the description at the PR should be updated.
> > @juanangp @lobis clang-formatting is only active for framework repository I guess.
> 
> Yes, I was just wondering this, I think @juanangp set it up to work with submodules too, maybe we should take a look.

It should work for submodules, but I just found that if `pre-commit` is found under `framework` it will skip the submodules. It means that the submodules will be skipped if you have performed `cmake` in between these 2 PR:
https://github.com/rest-for-physics/framework/pull/203
https://github.com/rest-for-physics/framework/pull/208
I will provide a solution.

**Comment by jgalan** [comment]
Then, I guess the solution is to `rm -r` your build directory and run `cmake` again?

**Comment by juanangp** [comment]
> Then, I guess the solution is to `rm -r` your build directory and run `cmake` again?

The git `hooks` are under `.git` folder, you should actually remove `.git/hooks/pre-commit` from the main folder in the repo. But a solution is already provided in https://github.com/rest-for-physics/framework/pull/227

File: inc/TRestGeant4Track.h [hunk 1]
```diff
@@ -114,7 +114,7 @@ class TRestGeant4Track : public TObject {
 
     inline Bool_t isRadiactiveDecay() const {
         for (int n = 0; n < fHits.GetNumberOfHits(); n++)
-            if (fHits.GetHitProcess(n) == 11) return true;
+            if ((fHits.GetHitProcess(n) == 11) ||(fHits.GetHitProcess(n) == 13)||(fHits.GetHitProcess(n) == 14))return true;
```
Comment (jgalan) [inline inc/TRestGeant4Track.h:117]:
```suggestion
            if ( fHits.GetHitProcess(n) == 11 || fHits.GetHitProcess(n) == 13 || fHits.GetHitProcess(n) == 14 ) return true;
```

**Pull request merged**